### PR TITLE
added status to model, added suspend endpoint, testing, linting

### LINKED ISSUE #25: Create Action route
As a Customer Manager
I need a route to perform stateful operations on the database
So that I can perform operations such as for example suspending a customer, etc.

**Details & Assumptions:**

- We will implement a new endpoint in routes
- The action should do something to the resource other than CRUD

**Acceptance Criteria:**

```
Given there is a customer database
When I access the endpoint
Then the stateful action should be performed
```

### PR DESCRIPTION
**Pull request by sncr0**
- added status to model
- added suspend endpoint
- testing
- linting

### CODE REVIEW

**Comment by sncr0** [comment]
do you guys think the post method I used in the route is correct? I also added it to the readme but without the post (given that is it non restful action)

File: tests/test_routes.py [hunk 1]
```diff
@@ -217,6 +217,30 @@ def test_delete_non_existing_customer(self):
...
+        new_customer_id = new_customer["id"]
+
+        response = self.client.post(f"/customers/{new_customer_id}/suspend")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        response = self.client.get(f"{BASE_URL}/{new_customer_id}")
+        suspended_customer = response.get_json()
+        self.assertEqual(suspended_customer["status"], 'suspended')
+
+    def test_suspend_non_existing_customer(self):
```
Comment (ch4262) [inline tests/test_routes.py:239]: Should this test go in the sad paths?

**Comment by ch4262** [comment]
We should also add the status attribute to the test_models.py to make sure those test cases account for the new attribute

**Comment by ch4262** [comment]
> do you guys think the post method I used in the route is correct? I also added it to the readme but without the post (given that is it non restful action)

I think we need something like this: https://github.com/nyu-devops/lab-flask-tdd/blob/9152b4b00336e50079fb3661cff2be384d2f8912/service/routes.py#L194

**Pull request merged**